src/Compilers/CSharp/Portable/Lowering/Instrumentation/DebugInfoInjector.cs

```diff
@@ -58,21 +58,65 @@ public override BoundStatement InstrumentExpressionStatement(BoundExpressionStat
 
             if (original.IsConstructorInitializer())
             {
-                switch (original.Syntax.Kind())
+                switch (original.Syntax)
                 {
-                    case SyntaxKind.ConstructorDeclaration:
-                        // This is an implicit constructor initializer.
-                        var decl = (ConstructorDeclarationSyntax)original.Syntax;
-                        return new BoundSequencePointWithSpan(decl, rewritten, CreateSpanForConstructorInitializer(decl));
-                    case SyntaxKind.BaseConstructorInitializer:
-                    case SyntaxKind.ThisConstructorInitializer:
-                        var init = (ConstructorInitializerSyntax)original.Syntax;
-                        Debug.Assert(init.Parent is object);
-                        return new BoundSequencePointWithSpan(init, rewritten, CreateSpanForConstructorInitializer((ConstructorDeclarationSyntax)init.Parent));
+                    case ConstructorDeclarationSyntax ctorDecl:
+                        // Implicit constructor initializer:
+                        return new BoundSequencePointWithSpan(ctorDecl, rewritten, createSpanForConstructorInitializer(ctorDecl));
```

Comment:
If we are expecting that the node always lacks an implicit constructor initializer, it might make sense to add an assertion, and also to split the local function `createSpanForConstructorInitializer`, into the code path which is hit for `case ConstructorDeclarationSyntax` here and the code path which is hit for `case ConstructorInitializerSyntax` below.